scripts/build-wda.js

```diff
@@ -1,27 +1,38 @@
 const {WebDriverAgent} = require('appium-webdriveragent');
 const xcode = require('appium-xcode');
-const B = require('bluebird');
 const {Simctl} = require('node-simctl');
 const {getSimulator} = require('appium-ios-simulator');
 const {logger} = require('appium/support');
 
 const log = logger.getLogger('WDA');
 
-// TODO: allow passing in all the various build params as CLI args
+function parseArgValue(argName) {
+  const argNamePattern = new RegExp(`^--${argName}\\b`);
+  for (let i = 1; i < process.argv.length; ++i) {
+    const arg = process.argv[i];
+    if (argNamePattern.test(arg)) {
+      return arg.includes('=') ? arg.split('=')[1] : process.argv[i + 1];
+    }
+  }
+  return null;
+}
+
 async function build() {
-  const [xcodeVersion, platformVersion] = await B.all([
-    xcode.getVersion(true),
-    xcode.getMaxIOSSDK(),
-  ]);
+  const customDevice = parseArgValue('name');
+  const xcodeVersion = await xcode.getVersion(true);
+  const platformVersion = parseArgValue('sdk') || (await xcode.getMaxIOSSDK());
+
   const verifyDevicePresence = (info) => {
     if (!info) {
-      throw new Error(`Cannot find any available iOS ${platformVersion} Simulator on your system`);
+      throw new Error(
+        `Cannot find any available iOS ${platformVersion} ${customDevice || ''} Simulator on your system`,
+      );
     }
     return info;
   };
   const deviceInfo = verifyDevicePresence(
     (await new Simctl().getDevices(platformVersion, 'iOS')).find(({name}) =>
-      name.includes('iPhone'),
+      name.includes(customDevice || 'iPhone'),
```

Comment:
I assume we should change the logic here to improve error messages and make them more helpful for users:
 - if the provided platform version is not present we must include all available versions in the error message
 - if the provided device name does not match for the given platform then we must include all device names for that platform

Reply:
Pushed, @mykola-mokhnach 

1) Platform doesn't exist, device not provided, there's already error wrapper for this
![image](https://github.com/user-attachments/assets/e11fe0f1-2344-4bdf-a7fe-bb724a4668f5)
2) Platform exists, but device is not correct
![image](https://github.com/user-attachments/assets/f4e14431-3cdf-4042-b136-eeebb7825ec1)
3) No platform, no device
![image](https://github.com/user-attachments/assets/07c2c5ac-de7b-431b-a57d-8462202b3f05)

Reply:
Android simulator o_O?